fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapper.kt

```diff
@@ -13,12 +13,25 @@ import org.mozilla.fenix.settings.SupportUtils
 /**
  * Returns a list of all the required Nimbus 'cards' that have been converted to [OnboardingPageUiData].
  */
-internal fun Collection<OnboardingCardData>.toPageUiData(showNotificationPage: Boolean): List<OnboardingPageUiData> =
+internal fun Collection<OnboardingCardData>.toPageUiData(
+    showNotificationPage: Boolean,
+    showAddWidgetPage: Boolean,
+    showSyncPage: Boolean,
+): List<OnboardingPageUiData> =
     filter {
-        if (it.cardType == OnboardingCardType.NOTIFICATION_PERMISSION) {
-            showNotificationPage
-        } else {
-            true
+        when (it.cardType) {
+            OnboardingCardType.NOTIFICATION_PERMISSION -> {
+                showNotificationPage
+            }
+            OnboardingCardType.ADD_SEARCH_WIDGET -> {
+                showAddWidgetPage
```

Comment:
As far as I can see, add search widget is only gated by the version check?

Reply:
Conditioned it by the new `enabled` flag and version check